### ISSUE TITLE
installation errors with fresh emacs 27  

### ISSUE THREAD

**Issue by malcook** [jkitchin/scimax#226]
Hi - I just built emacs under CentOS:
 
GNU Emacs 27.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.22.10)
 of 2018-08-20

and tried

   emacs -q -l path/to/scimax/init.el

To find:

Error (use-package): elpy/:catch: Symbol’s value as variable is void: xref-location-marker
Error (use-package): Failed to install helm-projectile: https://melpa.org/packages/projectile-20180815.2057.el: Not found
Error (use-package): Cannot load helm-projectile
Error (use-package): Failed to install magit: https://melpa.org/packages/magit-20180816.1819.tar: Not found
Error (use-package): Failed to install magithub: https://melpa.org/packages/magit-20180816.1819.tar: Not found
Error (use-package): Failed to install projectile: https://melpa.org/packages/projectile-20180815.2057.el: Not found
Error (use-package): scimax-notebook/:catch: Symbol’s function definition is void: projectile-mode

Any recommendations?  

Back out to emacs 26.x ??

Thanks!

~Malcolm

**Comment by jkitchin** [comment]
I have never tried an emacs 27 build. Many of those errors sound like melpa errors that may be resolved if you run package-refresh-contents, or on the next melpa build. 

I am a little surprised by the elpy/:catch error. If xref-location-marker is not in emacs27, that is probably not easily fixable, and going back to emacs 26 might be the thing to do. Unless you don't use elpy, and then just comment it out in packages.el

**Comment by malcook** [comment]
Thx.  package-refresh-contents fixed a few things, but I would up rebuilding scimax under emacs 26 anyway.  

I'll report new experience on this issue, if you don't mind.

scimax seemed to build OK, except:

   Error (use-package): Cannot load techela

Except , quiting scimax fials to quit.  When I  "C-x C-c"  I get a spinning cursor.  I have to `killall emacs` to quit the process!

I do the first the following message:

Wrote /home/mec/scimax/user/ac-comphist.dat

Any further advice from you would be very much wellcome!

**Comment by jkitchin** [comment]
Thanks for reporting these, the techela error is new today (unintended consequence of me removing the repo!). I think I have pushed some fixes for this issue to remove the submodule and dependency for now.

I am not sure what would cause the second issue.

**Comment by excetara2** [comment]
I just updated scimax to the newest version from a previous one. I am getting the techela error as well. I think it is still in the load scripts or something. Everytime I start up it tells me it can't load.

**Comment by jkitchin** [comment]
Do you see techela in your .gitmodules file?

**Comment by excetara2** [comment]
Yes, it was there. and the folder was made in the main directory.

I tried just recloning scimax but they were still there. THis is the error.

error: Server does not allow request for unadvertised object aaffbb2086455ebc3b94096393ee7eafa3fab78c
Fetched in submodule path 'techela', but it did not contain aaffbb2086455ebc3b94096393ee7eafa3fab78c. Direct fetching of that commit failed.

**Comment by jkitchin** [comment]
weird. that is not in https://github.com/jkitchin/scimax/blob/master/.gitmodules.

Are you cloning into a new directory, or running git pull in an old scimax directory?

**Comment by excetara2** [comment]
I cloned a new directory after moving the old one. I can try cloning again.

**Comment by jkitchin** [comment]
Are you using one of these commands at https://github.com/jkitchin/scimax#installation?

**Comment by excetara2** [comment]
Sorry major stupidity. I thought I was cloning your repo but I was cloning my fork which I hadn't rebased. It's compling the new files now so let you know if any problems.
